src/p2p/index.ts

```diff
@@ -89,18 +95,45 @@ export class P2PNetwork extends EventEmitter implements Service {
 
       });
 
+      // 2-way handshake
+      const protocol: string = "/eth/serenity/beacon/rpc/1";
+      const helloMsg: Hello = {
+       
+      };
+      this.node.handle(protocol, (proto, conn) => {
+        pull(
+          pull.values([Buffer.from(JSON.stringify(helloMsg))]),
+          conn,
+          pull.collect((values) => {
+	    // Peers' responses
+
+	  })
+	);	    
+      });
       this.node.on('peer:connect', (peerInfo) => {
         try {
           this.log.info(`Peer connected: ${peerInfo}`);
           this.peerBook.put(peerInfo);
-          this.discoveredPeers.add(peerInfo);
+	  this.discoveredPeers.add(peerInfo);
+	  this.node.dialProtocol(peerInfo, protocol, (err, conn) => {
+	    pull(
+              pull.values([Buffer.from(JSON.stringify(helloMsg))]),
+              conn,
+              pull.collect((values) => {
+	        // Peers responses
+
+	      })
+	    );
+	  })
+
         } catch (err) {
           this.log.error(err);
         }
       });
 
       this.node.on('peer:disconnect', (peerInfo) => {
-        try {
+	try {
+            
```

Comment:
white space, also not sure if this needs a `try/catch` theres no async functionality happening

Reply:
The `pull-stream` is there because that's what used in the example I was given. I will be changing this to an `async` iterator. Also, you can still use `try/catch` whether or not there's async functionality. Is there a preferred way for catching errors that would be better in this case?

Reply:
The way I look at It, those functions should return a promise or callback if they have an error.

I'm on mobile but I'll check the api later